app/src/main/java/com/example/taskyy/data/repositories/UserPreferencesImpl.kt

```diff
@@ -0,0 +1,57 @@
+package com.example.taskyy.data.repositories
+
+import android.content.SharedPreferences
+import com.example.taskyy.domain.objects.AuthenticatedUser
+import com.example.taskyy.domain.repository.UserPreferences
+import com.google.gson.Gson
+
+class UserPreferencesImpl(
+    private val sharedPref: SharedPreferences
+) : UserPreferences {
+    override fun saveAuthenticatedUser(user: AuthenticatedUser, key: String) {
+        val gson = Gson()
+        val json = gson.toJson(user)
+        sharedPref.edit().putString(key, json).apply()
+    }
+
+    override fun clearPreferences() {
+        sharedPref.edit().clear().apply()
+    }
+
+    override fun addUserFullName(fullName: String, key: String) {
+        sharedPref.edit().putString(fullName, key).apply()
+    }
+
+    override fun getUserFullName(key: String): String {
+        val name = sharedPref.getString(key, "")
+        return name!!
+    }
+
+    override fun addUserEmail(email: String, key: String) {
+        sharedPref.edit().putString(email, key).apply()
```

Comment:
I think the key comes first and then the email. That's likely why you're not getting it back